src/plugins/intel_npu/src/plugin/npuw/base_sync_infer_request.hpp

```diff
@@ -177,6 +191,12 @@ class IBaseInferRequest : public ov::ISyncInferRequest {
     using now_t = std::optional<std::size_t>;
     now_t now_idx() const;
 
+    std::vector<RListenerPtr> m_subscribers;
+    // IInferRequestSubmissionListener : dispatch impl
+    void subscribe_subrequest(std::size_t idx, Completed cb) override;
+    void complete_subrequest(ov::SoPtr<ov::IAsyncInferRequest> request, std::size_t idx) override;
```

Comment:
this ireq may be not needed, and not used right now, may be we need just list of tensors that are being produced in original model. GetTensor can be called as part of complete subrequest internally by IBaseInferRequest, or its detivatives